t3f/initializers.py

```diff
@@ -483,3 +483,79 @@ def zeros_like(tt):
       return matrix_zeros(shape)
     else:
       return tensor_zeros(shape[0, :])
+
+
+def random_matrix_xavier(shape, tt_rank=2):
+  """Constructs a random TT matrix with entrywise variance 2.0 / (n_in + n_out)
+
+  Args:
+    shape: 2d array, shape[0] is the shape of the matrix row-index,
+      shape[1] is the shape of the column index.
+      shape[0] and shape[1] should have the same number of elements (d)
+      Also supports ommiting one of the dimensions for vectors, e.g.
+        random_matrix([[2, 2, 2], None])
+      and
+        random_matrix_xavier([None, [2, 2, 2]])
+      will create an 8-element column and row vectors correspondingly.
+    tt_rank: a number.
+    mean: a number, the mean of the normal distribution used for
+      initializing TT-cores.
+    stddev: a number, the standard deviation of the normal distribution used
+      for initializing TT-cores.
+
+  Returns:
+    TensorTrain containing a TT-matrix of size
+      np.prod(shape[0]) x np.prod(shape[1])
+  """
+
+  # In case the shape is immutable.
+  shape = list(shape)
+  # In case shape represents a vector, e.g. [None, [2, 2, 2]]
+  if shape[0] is None:
+    shape[0] = np.ones(len(shape[1]), dtype=int)
+  # In case shape represents a vector, e.g. [[2, 2, 2], None]
+  if shape[1] is None:
+    shape[1] = np.ones(len(shape[0]), dtype=int)
+  shape = np.array(shape)
+  tt_rank = np.array(tt_rank)
+  if len(shape.shape) != 2:
+    raise ValueError('shape should be 2d array')
+  if shape[0].size != shape[1].size:
+    raise ValueError('shape[0] should have the same length as shape[1]')
+  if np.any(shape.flatten() < 1):
+    raise ValueError('all elements in `shape` should be positive')
+  if np.any(tt_rank < 1):
+    raise ValueError('`rank` should be positive')
+  if tt_rank.size != 1 and tt_rank.size != (shape[0].size + 1):
+    raise ValueError('`rank` array has inappropriate size')
+  if not all(isinstance(sh, np.integer) for sh in shape.flatten()):
+    raise ValueError('all elements in `shape` should be integers')
+  if tt_rank.size == 1:
+    if not isinstance(tt_rank[()], np.integer):
+      raise ValueError('`tt_rank` should be integer')
+  if tt_rank.size > 1:
+      raise ValueError('`tt_rank` should be a number')
+
+  num_dims = shape[0].size
+  if tt_rank.size == 1:
+    tt_rank = tt_rank * np.ones(num_dims - 1)
+    tt_rank = np.concatenate([[1], tt_rank, [1]])
+
+  n_in = np.prod(shape[0])
+  n_out = np.prod(shape[1])
+  lamb = 2.0 / (n_in + n_out)
+
+  # Empirically entries of TT-Matrix have variance r^(d-1)) and mean 0.
+  # We scale each TT-core to obtain the desired Xavier variance
+  # Var_xavier = 2 / (n_in + n_out)
+
+  sigma = (lamb * tt_rank[1] ** (1.0 - num_dims)) ** (1.0 / (2 * num_dims))
```

Comment:
Why TT-rank is a vector but you use only one element? Please, check that the input tt_rank is a number and throw a NotImplementedError otherwise